modules/ocs2/src/main/scala/gem/ocs2/pio/PioDecoder.scala

```diff
@@ -0,0 +1,39 @@
+package gem.ocs2.pio
+
+import PioError._
+
+import java.time.Instant
+
+import scala.xml.Node
+import scalaz.Scalaz._
+import scalaz._
+
+/** Typeclass for decoders of PIO-ish XML exported from an OCS2 database.
+  */
+trait PioDecoder[A] { outer =>
+  def decode(n: Node): PioError \/ A
+}
+
+object PioDecoder {
+  def apply[A](implicit ev: PioDecoder[A]): PioDecoder[A] = ev
+
+  def apply[A](f: Node => PioError \/ A): PioDecoder[A] =
+    new PioDecoder[A] {
+      def decode(n: Node): PioError \/ A =
+        f(n)
+    }
+
+  def enum[A](dataType: String)(m: (String, A)*): PioDecoder[A] =
+    fromParse[A](dataType) { m.toMap.lift }
+
+  def fromParse[A](dataType: String)(parse: PioParse[A]): PioDecoder[A] =
```

Comment:
Correct, I couldn't see another way of making it clear what it was you were trying to parse.  Though I suppose we could require `A: ClassTag` and read the runtime type to place in the error?  It seems like that might clean it up a bit.